lib/lrama/state.rb

```diff
@@ -142,5 +160,274 @@ def rr_conflicts
         conflict.type == :reduce_reduce
       end
     end
+
+    def propagate_lookaheads(next_state)
+      next_state.kernels.map {|item|
+        lookahead_sets =
+          if item.position == 1
+            goto_follow_set(item.lhs)
+          else
+            kernel = kernels.find {|k| k.predecessor_item_of?(item) }
```

Comment:
[note]: `C[k]` is the kernel (1) whose LHS and RHS are same with the item's one and (2) whose position is `item.position - 1`.